### ISSUE TITLE
presto-cli lack of support for http authentication

### ISSUE THREAD

**Issue by michaelcullum** [prestodb/presto#8361]
As a simpler way of adding authentication we have nginx sitting on the presto server which passes requests, does failover and handles http authentication; presto itself is fine with this, the only issues are with the Presto CLI client where it will ignore any http authentication credentials in the defined server.

e.g.
```
michael@mmp bin $ presto-cli --server https://username:password@madeuphostname.com --catalog hive --schema data_orc --user michael --source "My Local MBP"
presto:data_orc> show tables;
Error running command: Error starting query at https://madeuphostname.com/v1/statement returned an invalid response: JsonResponse{statusCode=401, statusMessage=Unauthorized, headers={Connection=[keep-alive], Server=[nginx/1.12.0], Content-Length=[195], Date=[Sat, 24 Jun 2017 21:36:58 GMT], Content-Type=[text/html], WWW-Authenticate=[Basic realm="Restricted"]}, hasValue=false, value=null} [Error: <html>
<head><title>401 Authorization Required</title></head>
<body bgcolor="white">
<center><h1>401 Authorization Required</h1></center>
<hr><center>nginx/1.12.0</center>
</body>
</html>
]
```

The other issue that might come up (un-tested) will be that if it follows `nextUri` statements they include the hostname which would be localhost. It would be a breaking change but generally it's my belief these should be relative, not absolute.

**Comment by electrum** [comment]
We intentionally dont allow specifying a password in the URL as it's easy for other processes to see the command line.

You can use --password to prompt for one or set the PRESTO_PASSWORD environment variable. User comes from --user.

**Comment by electrum** [comment]
Note that we also require HTTPS for basic auth, but it looks like you already have that.

**Comment by electrum** [comment]
We also just merged support for using an HTTP proxy (in addition to SOCKS) for the CLI and JDBC. Though it doesn't support specifying authentication, you need to use

-Dhttp.proxyUser=xxx
-Dhttp.proxyPassword=yyy

This may be a better option for you.

**Comment by michaelcullum** [comment]
@electrum I attempted to use the `--password` option but it would hang on queries (until i manually aborted, output in debug mode below)
```
presto:data_orc> show tables;
java.lang.RuntimeException: StatementClient thread was interrupted
	at com.facebook.presto.client.StatementClient.advance(StatementClient.java:280)
	at com.facebook.presto.cli.StatusPrinter.printInitialStatusUpdates(StatusPrinter.java:122)
	at com.facebook.presto.cli.Query.renderQueryOutput(Query.java:123)
	at com.facebook.presto.cli.Query.renderOutput(Query.java:107)
	at com.facebook.presto.cli.Console.process(Console.java:334)
	at com.facebook.presto.cli.Console.runConsole(Console.java:259)
	at com.facebook.presto.cli.Console.run(Console.java:145)
	at com.facebook.presto.cli.Presto.main(Presto.java:32)
Query aborted by user
```

With regards to the new arguments which PR was that? I can't see it and have just attempted locally and am getting unexpected argument errors so I'm guessing that it was merged before the most recent release?

**Comment by electrum** [comment]
The HTTP proxy support was just merged and will be in 0.180

**Comment by synhershko** [comment]
Should this be closed?
